components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -140,6 +157,32 @@ open class GleanInternalAPI internal constructor () {
         return uploadEnabled
     }
 
+    /**
+     * Clear any pending metrics when telemetry is disabled.
+     */
+    fun clearMetrics() = Dispatchers.API.launch {
+        // There is only one metric that we want to survive after clearing all metrics:
+        // firstRunDate. Here, we store its value so we can restore it after clearing
+        // the metrics.
+        val firstRunDateMetric = GleanInternalMetrics.firstRunDate
+        val existingFirstRunDate = DatetimesStorageEngine.getSnapshot(
+            firstRunDateMetric.sendInPings[0], false)?.get(firstRunDateMetric.identifier)?.let {
+            parseISOTimeString(it)
+        }
+
+        pingStorageEngine.clearPendingPings()
+        storageEngineManager.clearAllStores()
```

Comment:
Please note that this doesn't clear experiments: https://github.com/mozilla-mobile/android-components/blob/81e5463966ed6115c4707350059fa859574ecbbf/components/service/glean/src/main/java/mozilla/components/service/glean/storages/StorageEngineManager.kt#L26

Reply:
Good catch.  Do we want to clear experiments?  I'm a bit concerned about weird interactions with the experiments library if we do, particularly around *restoring* the right state if we turn uploading back on.

Reply:
Cc: @georgf ?

Reply:
I *think* that it shouldn't matter: this is only tagging for glean pings. If we turn upload off, then experiments or not, we don't send anything.

Reply:
I think I'll add a comment for now saying that this doesn't clear experiments on purpose.